Helper/WebHookHandlers/ChargeCanceled.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Vindi\Payment\Helper\WebHookHandlers;
+
+class ChargeCanceled
+{
+    /**
+     * @var \Vindi\Payment\Model\Payment\Bill
+     */
+    protected $bill;
+
+    /**
+     * @var \Vindi\Payment\Helper\WebHookHandlers\Order
+     */
+    protected $order;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @var \Magento\Sales\Api\OrderRepositoryInterface
+     */
+    protected $orderRepository;
+
+    public function __construct(
+        \Magento\Sales\Api\OrderRepositoryInterface $orderRepository,
+        \Vindi\Payment\Model\Payment\Bill $bill,
+        \Vindi\Payment\Helper\WebHookHandlers\Order $order,
+        \Psr\Log\LoggerInterface $logger
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->bill = $bill;
+        $this->order = $order;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param array $data
+     *
+     * @return bool
+     * @throws \Exception
+     */
+    public function chargeCanceled($data)
+    {
+        $charge = $data['charge'];
+
+        /** @var \Magento\Sales\Model\Order $order */
+        if (!($order = $this->getOrderFromBill($charge['bill']['id']))) {
+            $this->logger->warning(__('Order not found'));
+
+            return false;
+        }
+
+        $order->cancel();
+        $order->addStatusHistoryComment(__(sprintf('Vindi API: Order Canceled.')));
```

Comment:
```suggestion
        $order->addStatusHistoryComment(__(sprintf(
            'Vindi API: Order %s Canceled.',
            $order->getId()
        )));
```
O que acha de colocar também o ID do pedido no histórico?

Reply:
Feito ajuste.